src/profiler.cpp

```diff
@@ -149,8 +149,7 @@ void Profiler::checkDeadline() {
         ssize_t w = write(STDERR_FILENO, error, sizeof(error) - 1);
         (void) w;
 
-        _running = false;
-        setTimer(0, 0);
+        stop();
```

Comment:
Right. I'll get rid of the whole deadline thing for now, then, and create a separate issue to track it, as you suggested.